src/main/java/org/osc/controller/nsc/restserver/api/InspectionApis.java

```diff
@@ -0,0 +1,403 @@
+        }
+        LOG.info(String.format("Creating inspection port for (ingress id %s ; egress id %s)",
+                "" + entity.getIngressPort().getElementId(), "" + entity.getEgressPort().getElementId()));
+        init(ipAddr);
+        return this.txControl.required(() -> {
+
+            InspectionPortEntity inspectionPortEntity = (InspectionPortEntity) this.api.getInspectionPort(entity);
+
+            if (inspectionPortEntity != null) {
+                String msg = String.format("inspection port element already exists id: %s\n", entity.getElementId());
+                LOG.error(msg);
+                throw new IllegalArgumentException(msg);
+            }
+            inspectionPortEntity = this.utils.makeInspectionPortEntity(entity);
+            return this.em.merge(inspectionPortEntity).getElementId();
+        });
+    }
+
+    @Path("/ipAddr/{ipAddr}/insport/{portId}")
+    @PUT
+    public InspectionPortEntity updateInspectionPort(@PathParam("ipAddr") String ipAddr,
+            @PathParam("portId") String portId, InspectionPortEntity entity) throws Exception {
+
+        LOG.info(String.format("Updating the inspection port element id %s ", "" + portId));
+        init(ipAddr);
+        return this.txControl.required(() -> {
+
+            InspectionPortEntity inspectionPortEntity = (InspectionPortEntity) this.api.getInspectionPort(entity);
+
+            if (inspectionPortEntity == null) {
+                String msg = String.format("inspection port element does not exists id: %s\n", entity.getElementId());
+                LOG.error(msg);
+                throw new IllegalArgumentException(msg);
+            }
+            inspectionPortEntity = this.utils.makeInspectionPortEntity(entity);
+            return this.em.merge(inspectionPortEntity);
+        });
+    }
+
+    @Path("/ipAddr/{ipAddr}/insport/{portId}")
+    @DELETE
+    public void deleteInspectionPort(@PathParam("ipAddr") String ipAddr, @PathParam("portId") String portId)
+            throws Exception {
+        if (portId == null) {
+            throw new IllegalArgumentException("Attempt to update null inspection port");
+        }
+        LOG.info(String.format("Deleting the inspection port element for id %s ", "" + portId));
+        init(ipAddr);
+        InspectionPortEntity inspectionPort = new InspectionPortEntity(portId, null, null);
+        this.api.removeInspectionPort(inspectionPort);
+    }
+
+    @Path("/ipAddr/{ipAddr}/insport")
+    @GET
+    public List<String> getInspectionPortIds(@PathParam("ipAddr") String ipAddr) throws Exception {
+        LOG.info("Listing inspection port ids'");
+        init(ipAddr);
+
+        return this.txControl.supports(() -> {
+            List<? extends InspectionPortEntity> elements = this.utils.txInspectionPortEntities();
+            List<String> elementList = new ArrayList<String>();
+            if (elements.isEmpty()) {
```

Comment:
revert the if to eliminate one of the returns

Reply:
ok